### PR TITLE
[NEW] Livechat agent communication disable switch

### PR DESCRIPTION
**Pull request by AlexanderKanakis**

### CODE REVIEW

File: app/livechat/lib/LivechatRoomType.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ export default class LivechatRoomType extends RoomTypeConfig {
 	}
 
 	condition() {
+		console.log('in');
```
Comment (itzzritik) [inline app/livechat/lib/LivechatRoomType.js]: Why this console.log statement?

Comment (AlexanderKanakis) [reply app/livechat/lib/LivechatRoomType.js]: Removed unnecessary log.

File: app/livechat/server/hooks/beforeDelegateAgent.js [hunk 2]
```diff
@@ -5,18 +5,20 @@ import { Users, LivechatDepartmentAgents } from '../../../models';
 
 callbacks.add('livechat.beforeDelegateAgent', (options = {}) => {
 	const { department, agent } = options;
+	if (!settings.get('Livechat_kill_switch')) {
```
Comment (itzzritik) [inline app/livechat/server/hooks/beforeDelegateAgent.js:8]: ### How about reducing code blocks nesting _(return early)_?

sample code for reference:

```
if (settings.get('Livechat_kill_switch')) {
	return;
}
if (agent) {
	return agent;
}
if (!settings.get('Livechat_assign_new_conversation_to_bot')) {
	return;
}
if (department) {
	return LivechatDepartmentAgents.getNextBotForDepartment(department);
}
return Users.getNextBotAgent();
```

File: app/livechat/server/lib/routing/AutoSelection.js [hunk 3]
```diff
@@ -20,11 +21,13 @@ class AutoSelection {
 	}
 
 	getNextAgent(department, ignoreAgentId) {
-		if (department) {
-			return LivechatDepartmentAgents.getNextAgentForDepartment(department, ignoreAgentId);
-		}
+		if (!settings.get('Livechat_kill_switch')) {
```
Comment (itzzritik) [inline app/livechat/server/lib/routing/AutoSelection.js:24]: ### How about an early return here as well?

sample code for reference:

```
if (settings.get('Livechat_kill_switch')) {
	return;
}
if (department) {
	return LivechatDepartmentAgents.getNextAgentForDepartment(department, ignoreAgentId);
}
return Users.getNextAgent(ignoreAgentId);
```

Comment (AlexanderKanakis) [reply app/livechat/server/lib/routing/AutoSelection.js:24]: Optimized code blocks.

**Review by chadgoss — APPROVED** [review]

**Pull request merged**